core/src/main/scala/org/apache/spark/api/python/PythonWorkerFactory.scala

```diff
@@ -95,10 +95,19 @@ private[spark] class PythonWorkerFactory(
   def create(): (PythonWorker, Option[Long]) = {
```

Comment:
Do you know what happens in the case where the worker dies right after the `create()` call returns successfully? I think we are rightfully solving one of the cases here where an idle worker sitting in the `idleWorkers` has died. The other case can still happen

Reply:
There's a bit of a timing question there, because the factory does not wait for a first bits to be sent to the worker. So this would result in an IO error still. We could add a first bit to be sent by the factory to the forked of process to signal aliveness though to capture this?

Reply:
I see. I think that might be an overkill. This is good enough IMO